asserts/account_key_test.go

```diff
@@ -590,6 +590,175 @@ func (aks *accountKeySuite) TestPublicKeyIsValidAt(c *C) {
+		"name: default\n" +
+		"public-key-sha3-384: " + aks.keyID + "\n" +
+		aks.sinceLine +
+		aks.untilLine +
+		fmt.Sprintf("body-length: %v", len(aks.pubKeyBody)) + "\n" +
+		"sign-key-sha3-384: Jv8_JiHiIzJVcO9M55pPdqSDWUvuhfDIBJUS-3VW7F_idjix7Ffn5qMxB21ZQuij" + "\n\n" +
+		aks.pubKeyBody + "\n\n" +
+		"AXNpZw=="
+	a, err := asserts.Decode([]byte(encoded))
+	c.Assert(err, IsNil)
+
+	accKey := a.(*asserts.AccountKey)
+
+	z := time.Time{}
+
+	tests := []struct {
+		earliest time.Time
+		latest   time.Time
+		valid    bool
+	}{
+		{aks.since, aks.until, true},
+		{aks.since, aks.since.AddDate(0, 3, 0), true},
+		{aks.since.AddDate(0, 1, 0), aks.since.AddDate(0, 3, 0), true},
+		{aks.since.AddDate(0, 1, 0), aks.until, true},
+		{aks.until, aks.until.AddDate(0, 3, 0), false},
+		{aks.until.AddDate(0, 2, 0), aks.until.AddDate(0, 3, 0), false},
+		{aks.since.AddDate(0, -1, 0), aks.since, true},
+		{aks.since.AddDate(0, -1, 0), aks.since.AddDate(0, 1, 0), true},
+		{aks.since.AddDate(0, -2, 0), aks.since.AddDate(0, -2, 0), false},
+		{aks.until.AddDate(0, -1, 0), aks.until.AddDate(0, 1, 0), true},
+		{aks.since, z, true},
+		{aks.since.AddDate(0, 1, 0), z, true},
+		{aks.since.AddDate(0, -3, 0), z, true},
+		{aks.until, z, false},
+		{aks.until.AddDate(0, 1, 0), z, false},
+	}
+
+	for _, t := range tests {
+		c.Check(asserts.AccountKeyIsKeyValidAssumingCurTimeWithin(accKey, t.earliest, t.latest), Equals, t.valid)
+	}
+
+}
+
+func (aks *accountKeySuite) TestPublicKeyIsValidAssumingCurTimeWithinNoUntilInterval(c *C) {
+	// With since and until, i.e. signing account-key expires.
+	// Key is valid for time >= since.
+	encoded := "type: account-key\n" +
+		"authority-id: canonical\n" +
+		"account-id: acc-id1\n" +
+		"name: default\n" +
+		"public-key-sha3-384: " + aks.keyID + "\n" +
+		aks.sinceLine +
+		fmt.Sprintf("body-length: %v", len(aks.pubKeyBody)) + "\n" +
+		"sign-key-sha3-384: Jv8_JiHiIzJVcO9M55pPdqSDWUvuhfDIBJUS-3VW7F_idjix7Ffn5qMxB21ZQuij" + "\n\n" +
+		aks.pubKeyBody + "\n\n" +
+		"AXNpZw=="
+	a, err := asserts.Decode([]byte(encoded))
+	c.Assert(err, IsNil)
+
+	accKey := a.(*asserts.AccountKey)
+
+	z := time.Time{}
+	later := aks.until
+
+	tests := []struct {
+		earliest time.Time
+		latest   time.Time
+		valid    bool
+	}{
+		{aks.since, later, true},
+		{aks.since, aks.since.AddDate(0, 3, 0), true},
+		{aks.since.AddDate(0, 1, 0), aks.since.AddDate(0, 3, 0), true},
+		{aks.since.AddDate(0, 1, 0), later, true},
+		{later, later.AddDate(0, 3, 0), true},
+		{later.AddDate(0, 2, 0), later.AddDate(0, 3, 0), true},
+		{aks.since.AddDate(0, -1, 0), aks.since, true},
+		{aks.since.AddDate(0, -1, 0), aks.since.AddDate(0, 1, 0), true},
+		{aks.since.AddDate(0, -2, 0), aks.since.AddDate(0, -2, 0), false},
+		{later.AddDate(0, -1, 0), later.AddDate(0, 1, 0), true},
+		{aks.since, z, true},
+		{aks.since.AddDate(0, 1, 0), z, true},
+		{aks.since.AddDate(0, -3, 0), z, true},
+		{later, z, true},
+		{later.AddDate(0, 1, 0), z, true},
```

Comment:
same as above I think we need tests here with `z` as the earliest parameter

Reply:
added some